react/BaseContent.tsx

```diff
@@ -26,11 +26,15 @@ const Content: FC<Props> = ({ finishShoppingButtonLink }) => {
   const { orderForm, loading }: OrderFormContext = useOrderForm()
   const handles = useCssHandles(CSS_HANDLES)
   const { variation } = useMinicartState()
-  const { url: checkoutUrl } = useCheckoutURL()
+  const { url: checkoutUrl, major } = useCheckoutURL()
   const { navigate } = useRuntime()
 
   const goToCheckout = (url: string) => {
-    navigate({ to: url })
+    if (major) {
+      navigate({ to: url })
+    } else {
+      window.location.href = url
+    }
```

Comment:
You should also take into account the case where `finishShoppingButtonLink` is truthy and the account has the new Cart installed. In this case, I believe you should not use `navigate` either. In short, you should only use `navigate` if it sends the user to the new Cart.